### ISSUE TITLE
client: Expose free and locked balances for an account

### ISSUE THREAD

**Issue by NunoAlexandre** [radicle-dev/radicle-registry#566]
Part of #530 
Depends on #569 

We need to expose not only the `free` balance of an account but also the total locked amount in deposits. 

- [ ] Add client API call to get a type gathering both values (or change current one, `free_balance`)
- [ ] Update the CLI to print both values
- [ ] Integrate with the upstream

**Comment by CodeSandwich** [comment]
Slashing is a financial punishment for misbehaving on a PoS network. If you build an invalid block, the network will agree to reject it and slash your staked value. We don't use it, at least not now, because we're using PoW and punishment for mining a bad block is simple rejection and wasting of your computation resources.

**Comment by NunoAlexandre** [comment]
Thanks, @CodeSandwich.

I would also appreciate yours and @geigerzaehler's thoughts on the proposed solutions.

**Comment by CodeSandwich** [comment]
We should also keep in mind that if we decide that the value of all each deposit is arbitrarily known, we may not need to implement that at all. I don't think that we've discarded this approach yet.

**Comment by NunoAlexandre** [comment]
> We should also keep in mind that if we decide that the value of all each deposit is arbitrarily known, we may not need to implement that at all. I don't think that we've discarded this approach yet.

Can you expand? I am not sure I understand.

**Comment by CodeSandwich** [comment]
If we arbitrarily know that e.g. an org deposit is always 100 RAD, we don't need to store the exact deposit amount, we can calculate it on the fly based on number of "owned" (reimbursement eligible) orgs.

**Comment by NunoAlexandre** [comment]
> If we arbitrarily know that e.g. an org deposit is always 100 RAD, we don't need to store the exact deposit amount, we can calculate it on the fly based on number of "owned" (reimbursement eligible) orgs.

Right. Is that assuming that the deposit amount for registering an org never changes? If so, that's not a way to go since we want to be able to change those amounts. Another alternative, that you possibly mean as well, is to reimburse the deposit based on what those amounts are in the current time. In that case, we go back to the discussion around what happens when that amount is different than the original locked amount. In a conversation with @cloudhead we agreed that reimbursing the originally locked amount is best, which - given we want the flexibility to change the deposit values - would require implementing this.

**Comment by CodeSandwich** [comment]
In the case of variable deposit amounts having a locked balance on the user side is not a good solution either. If user A owns orgs X and Y and has 210 RAD locked, how mush will he get back if he unregisters org X? The deposit should be written on the org side, this way we know that X has 110 RAD locked, but Y only 100. Having a locked balance counter on the user side would be duplication of information, which is always problematic.

**Comment by NunoAlexandre** [comment]
> In the case of variable deposit amounts having a locked balance on the user side is not a good solution either. If user A owns orgs X and Y and has 210 RAD locked, how much will he get back if he unregisters org X? The deposit should be written on the org side, this way we know that X has 110 RAD locked, but Y only 100. Having a locked balance counter on the user side would be duplication of information, which is always problematic.

Who is talking about users?

**Comment by CodeSandwich** [comment]
Ah, it's about an account attached to the org, not to the user? Sorry, I didn't catch that.

Are we planning to have deposits on project registrations as well?

**Comment by NunoAlexandre** [comment]
> Ah, it's about an account attached to the org, not to the user? Sorry, I didn't catch that.

This specific issue is not so much about to which entity the deposits are associated with but more about how to design the free <> locked balances. That is agnostic to which account it refers to. And no worries :ok_hand: 

> 
> Are we planning to have deposits on project registrations as well?

Yes, for both user and org project registrations.

**Comment by CodeSandwich** [comment]
> Yes, for both user and org project registrations.

Then neither solution A nor B will cover projects, they don't have accounts. Unless we add them, we could generate them on creation like in orgs.

**Comment by NunoAlexandre** [comment]
> > Yes, for both user and org project registrations.
> 
> Then neither solution A nor B will cover projects, they don't have accounts. Unless we add them, we could generate them on creation like in orgs.

The deposit for a project will be under its domain's (user or org) account. I don't follow how that is an issue to discuss the higher-level design approach? :thinking:

**Comment by CodeSandwich** [comment]
> The deposit for a project will be under its domain's (user or org) account.
Then we're back at the "user owns two projects, how much will he get back when one is unregistered" problem.

> I don't follow how that is an issue to discuss the higher-level design approach? thinking
This design has a specific problem to solve, we need to be sure that it actually will do that.

**Comment by NunoAlexandre** [comment]
> > The deposit for a project will be under its domain's (user or org) account.
> 
> Then we're back at the "user owns two projects, how much will he get back when one is unregistered" problem.

Hm not necessarily, that depends on the solution chosen for keeping track of that. My preference, to meet the requirements and to meet a good user experience, would be to go with option B with a fine granularity level, where we keep track of each locked deposit, to which @cloudhead emphasized on the storage costs.

This seems to be a point we get back into so I will create a dedicated issue to focus on that particular part.

**Comment by NunoAlexandre** [comment]
@CodeSandwich rephrased and re-scoped this issue and created a separate one: #569

**Comment by CodeSandwich** [comment]
We've ditched the deposits.
